cmd/gossamer/configcmd.go

```diff
@@ -273,7 +273,8 @@ func createNetworkService(fig *cfg.Config, gendata *genesis.Data, stateService *
 func createCoreService(coreConfig *core.Config) *core.Service {
 	coreService, err := core.NewService(coreConfig)
 	if err != nil {
-		log.Error("Failed to create new core service", "err", err)
+		log.Crit("Failed to create new core service", "err", err)
+		os.Exit(0)
```

Comment:
would that not be exit(1) since it was a fatal err ?

Reply:
well, but that was a fatal system error, since the core was not able to start and that is a critical error. If we exit with 0, it looks like it was a clean exit.

Reply:
yes, it did exit cleanly, because the program behaved as expected- we couldn't start a service so we exited. the error was "caught". this is not the same as a segfault or other memory access error, where the program is forced to exit by the kernel.

Reply:
the program failed to createCoreService because create it could not instantiate a new BABE session, that is not behaving as expected. for example: BDD: `Given a user set cfg.IsBabeAuthority to true, When a service fails to be instantiated, Then I  expect a exit with non zero `. This behavior is already present if you consider that main.go already exit with 1 when a err is returned. This is there so the ops knows something is wrong.

Since you wrote a critical log there and then assuming this is a critical error, I believe the logic thing to do is to propagate it back to the root (func createCoreService->func makeNode->func gossamer->exit(1) ) and that will exit the program with the propagated err.
https://github.com/ChainSafe/gossamer/blob/development/cmd/gossamer/main.go#L131